### ISSUE TITLE
Make Overhangs printable and printing Multi color adds color all the way down

### ISSUE THREAD

**Issue by Aghost917** [SoftFever/OrcaSlicer#4559]
### Is there an existing issue for this problem?

- [X] I have searched the existing issues

### OrcaSlicer Version

2.0.0-Beta

### Operating System (OS)

Windows

### OS Version

Windows 11 22H2

### Printer

Bambu Labs P1S

### How to reproduce

Import the STL file, use the paint tool on the STL, and turn on Make Overhangs Printable.

### Actual results

For some reason, it just adds color all the way to the build plate instead of starting at the layer height of the color change.
![image](https://github.com/SoftFever/OrcaSlicer/assets/36266586/d3ca091b-c9a4-4bbd-9c04-5a28b9ea0f59)

### Expected results

![image](https://github.com/SoftFever/OrcaSlicer/assets/36266586/7e4795ed-c6eb-4f9e-9797-bdaacb299626)

### Project file & Debug log uploads

[Rubber_Duck.zip](https://github.com/SoftFever/OrcaSlicer/files/14641051/Rubber_Duck.zip)

### Checklist of files to include

- [x] Log file
- [x] Project file

**Comment by okirisx2** [comment]
Yes. This has been a long-standing bug. I’ve not found a workaround.

**Comment by tomzdotorg** [comment]
`this->apply_conical_overhang();`

has to be above 

`apply_mm_segmentation(...`

in PrintObjectSlice.cpp

i.e. 

```
    this->apply_conical_overhang();

    // Is any ModelVolume MMU painted?
    if (const auto& volumes = this->model_object()->volumes;
        m_print->config().filament_diameter.size() > 1 && // BBS
        std::find_if(volumes.begin(), volumes.end(), [](const ModelVolume* v) { return !v->mmu_segmentation_facets.empty(); }) != volumes.end()) {

        // If XY Size compensation is also enabled, notify the user that XY Size compensation
        // would not be used because the object is multi-material painted.
        if (m_config.xy_hole_compensation.value != 0.f || m_config.xy_contour_compensation.value != 0.f) {
            this->active_step_add_warning(
                PrintStateBase::WarningLevel::CRITICAL,
                L("An object's XY size compensation will not be used because it is also color-painted.\nXY Size "
                  "compensation can not be combined with color-painting."));
            BOOST_LOG_TRIVIAL(info) << "xy compensation will not work for object " << this->model_object()->name << " for multi filament.";
        }

        BOOST_LOG_TRIVIAL(debug) << "Slicing volumes - MMU segmentation";
        apply_mm_segmentation(*this, [print]() { print->throw_if_canceled(); });
    }

```

**Comment by GarlicDebug** [comment]
I just attempted the fix mentioned by @tomzdotorg and it didn't work for the model I tested with.

**Comment by JamieWritesCode** [comment]
@SoftFever Can we get this re-opened and looked at again? This is still an issue as of today.

**Comment by tomzdotorg** [comment]
> This is still an issue as of today.

I believe the fix I posted above was merged in #6896

**Comment by JamieWritesCode** [comment]
Awesome, great stuff! Apologies for the unnecessary tag. I must have missed that PR when searching.
